### PR TITLE
Developing the complex VNA analysis

### PR DESCRIPTION
**Pull request by fluthi**
Please use the following template for a pull request. 

Changes proposed in this pull request:
- This is to get the MAv2 working for the standard (non-complex) VNA fitting
- Changed the timestamp extraction in the options dict of MAv2
- Additional wisdom will follow

In order for the pull request to be merged, the following conditions must be met:
- travis test suite passes
- all reasonable issues raised by codacy must be resolved 
- a positive review is required

Whenever possible the pull request should
- follow the PEP8 style guide 
- have tests for the code
- be well documented and contain comments

Tests are not mandatory as this is generally hard to make for instruments that interact with hardware.

### CODE REVIEW

File: pycqed/analysis/fitting_models.py [hunk 1]
```diff
@@ -310,33 +312,10 @@ def HangerFuncAmplitude(f, f0, Q, Qe, A, theta):
     return abs(A * (1. - Q / Qe * np.exp(1.j * theta) / (1. + 2.j * Q * (f / 1.e9 - f0) / f0)))
 
 
-def HangerFuncComplex(f, pars):
```
Comment (AdriaanRol) [inline pycqed/analysis/fitting_models.py]: I'm assuming this is being deleted on purpose?

File: pycqed/analysis_v2/base_analysis.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 import h5py
 from pycqed.measurement.hdf5_data import write_dict_to_hdf5
 
+from importlib import reload  # Useful for reloading while testing
```
Comment (AdriaanRol) [inline pycqed/analysis_v2/base_analysis.py]: Please remove this from base analysis.

File: pycqed/analysis_v2/base_analysis.py [hunk 3]
```diff
@@ -138,11 +140,14 @@ def __init__(self, t_start: str = None, t_stop: str = None,
         # These options determine what data to extract #
         ################################################
         scan_label = self.options_dict.get('scan_label', label)
+        if scan_label is None:
```
Comment (AdriaanRol) [inline pycqed/analysis_v2/base_analysis.py]: Would be much better to instead replace the default argument with `''`, a hidden if statement that replaces some parameter is quite horrible for debugging.

File: pycqed/analysis_v2/base_analysis.py [hunk 4]
```diff
@@ -224,7 +231,7 @@ def run_analysis(self):
             self.plot(key_list='auto')  # make the plots
 
         if self.options_dict.get('save_figs', False):
-            self.save_figures(close_figs=self.options_dict.get('close_figs', False))
+            self.save_figures(close_figs=self.options_dict.get('close_figs', True))
```
Comment (AdriaanRol) [inline pycqed/analysis_v2/base_analysis.py]: good catch 👍

File: pycqed/analysis_v2/base_analysis.py [hunk 5]
```diff
@@ -470,6 +477,7 @@ def prepare_fitting(self):
         # initialize everything to an empty dict if not overwritten
         self.fit_dicts = OrderedDict()
 
+
     def run_fitting(self):
```
Comment (AdriaanRol) [inline pycqed/analysis_v2/base_analysis.py]: Danger! I think this piece of code is getting awfully complicated to understand (as you have probably experienced when debugging the tests) I am sure the tests are working now but I'm a bit hesitant to approve this change... Let's discuss a bit, I'm not sure what the best solution is.

File: pycqed/analysis_v2/spectroscopy_analysis.py [hunk 6]
```diff
@@ -13,8 +13,11 @@
 from pycqed.analysis import fitting_models as fit_mods
 import lmfit
 
+from importlib import reload  # Useful for reloading while testing
```
Comment (AdriaanRol) [inline pycqed/analysis_v2/spectroscopy_analysis.py]: Please remove debugging stuff from here.

File: pycqed/analysis_v2/spectroscopy_analysis.py [hunk 7]
```diff
@@ -13,8 +13,11 @@
...
+
 import importlib
```
Comment (AdriaanRol) [inline pycqed/analysis_v2/spectroscopy_analysis.py]: Especially if it is imported multiple times.

File: pycqed/analysis_v2/spectroscopy_analysis.py [hunk 8]
```diff
@@ -210,22 +215,22 @@ def process_data(self):
         self.proc_data_dict['amp_label'] = 'Transmission amplitude (V rms)'
         self.proc_data_dict['phase_label'] = 'Transmission phase (degrees)'
         if len(self.raw_data_dict['timestamps']) == 1:
-            self.proc_data_dict['plot_phase'] = np.unwrap(np.pi / 180. * self.proc_data_dict['plot_phase']) * 180 / np.pi
+            self.proc_data_dict['plot_phase'] = np.unwrap(self.proc_data_dict['plot_phase'],discont=3.141592653589793)
```
Comment (AdriaanRol) [inline pycqed/analysis_v2/spectroscopy_analysis.py]: why not `np.pi` instead of 3.14159....

File: pycqed/analysis/fitting_models.py (hunk 1 above)
Comment (fluthi) [reply pycqed/analysis/fitting_models.py]: The standard complex hanger function should, just like the standard for the fitting models, take floats as arguments.
In order no to loose functionality the function taking a list of parameters as been renamed to hanger_func_complex_SI_pars.
I've added a comment to this.

File: pycqed/analysis_v2/base_analysis.py (hunk 2 above)
Comment (fluthi) [reply pycqed/analysis_v2/base_analysis.py]: Done.

File: pycqed/analysis_v2/base_analysis.py (hunk 3 above)
Comment (fluthi) [reply pycqed/analysis_v2/base_analysis.py]: The default argument is indeed a str='', see definition of the __init__ of the base_class.

File: pycqed/analysis_v2/spectroscopy_analysis.py (hunk 6 above)
Comment (fluthi) [reply pycqed/analysis_v2/spectroscopy_analysis.py]: Removed all the reload commands, also to reload the base_class and the fitting_mods

File: pycqed/analysis_v2/base_analysis.py (hunk 3 above)
Comment (AdriaanRol) [reply pycqed/analysis_v2/base_analysis.py]: if you agree, could you remove this addition?

File: pycqed/analysis_v2/spectroscopy_analysis.py (hunk 8 above)
Comment (fluthi) [reply pycqed/analysis_v2/spectroscopy_analysis.py]: Changed

**Comment by AdriaanRol** [comment]
Hi @fluthi you comment that the changes have been made but there are no new commits. Could you push the changes?

File: pycqed/analysis_v2/base_analysis.py (hunk 3 above)
Comment (fluthi) [reply pycqed/analysis_v2/base_analysis.py]: Tests fail without it. 
Cleanup of this should be addressed in a different pull request. So I'm going to change it back.

Comment (fluthi) [reply pycqed/analysis_v2/base_analysis.py]: Culprit is the default label in the spec analysis. This is now changed, and the if statement removed.
So should be all good now.

**Comment by fluthi** [comment]
Merged and deleted

**Pull request merged**